### DISCUSSION TITLE
How to change mime-type of a static file downloaded from App_Data/Sites/Default/wwwroot?

### DISCUSSION THREAD

**Discussion by devlife** [OrchardCMS/OrchardCore#16600] [Q&A]
I have a file with a custom file extension that I need to change the mime-type when it's downloaded from `OC Host/App_Data/Sites/Default/wwwroot/my.file.customextension`. Right now it gets downloaded as applicaiton/octet-stream. 

I've tried adding mime type mappings to web.config as well as putting the following in my theme startup

```csharp
app.UseStaticFiles(new StaticFileOptions()
{
    ContentTypeProvider = new FileExtensionContentTypeProvider(new Dictionary<string, string>
    {
    　 { 
         ".customextension",
         "application/json"
       }
    })
});
```

Any thoughts would be welcome.

**Comment by Piedone** [comment]
While I don't know for sure, my first take would be to use `StaticFileOptions` too. However, what you have might be overwritten later. Try to configure it with `services.Configure<StaticFileOptions>(o => ...);` from a module instead, since [that'll be used](https://github.com/OrchardCMS/OrchardCore/blob/bf42ea903bcc09c697d1da4b33e8f18ecccf00b5/src/OrchardCore/OrchardCore/Modules/Extensions/ServiceCollectionExtensions.cs#L260).

**Comment by weirdyang** [comment]
something like this?
![image](https://github.com/user-attachments/assets/dca6ff01-1e0c-4a17-9ca5-6cfbeaea6b5d)
```
            var provider = new FileExtensionContentTypeProvider(new Dictionary<string, string>
            {
                {
                    ".xxx",
                    "text/plain"
                }
            });
            app.UseStaticFiles(new StaticFileOptions
            {
                ContentTypeProvider = provider,
                ServeUnknownFileTypes = true,
                DefaultContentType = "application/octet-stream",
            });
```
change the "text/plain" to what you need, eg. "application/wasm"
![image](https://github.com/user-attachments/assets/a3b49ab9-d9f9-4b55-a509-1c8c777b755b)

**Reply by devlife** [reply]
Where, specifically, did you put the file and where did you put the code? I tried a bunch of options but couldn't get it to work.

**Reply by weirdyang** [reply]
File is in the wwwroot folder and the code is in the main application program.cs

**Reply by weirdyang** [reply]
> Where, specifically, did you put the file and where did you put the code? I tried a bunch of options but couldn't get it to work.

@devlife 
![image](https://github.com/user-attachments/assets/f8e5a3cc-da98-43f3-967d-7ec83c409cc2)
